### ISSUE TITLE
Establish maximum mentor-to-student ratio for individual projects

### ISSUE THREAD

**Issue by aeksco** [rcos/rcos-handbook#58]
**Description of Issue:** We can't have one mentor managing 10 students on a single project. It makes small group meetings incredibly difficult to manage and students in need of help can't get sufficient attention.

**Proposed Solution:** We establish a maximum mentor-to-student ratio for individual projects. I propose 1 mentor to 5 students as the maximum allowed ratio without specific permission.

@AdrianCollado @wdturner @Bad-Science @kburk1997 Thoughts?

**Comment by kburk1997** [comment]
@Bad-Science and I were in a small group with mentors mainly from YACS, and this definitely made it difficult for us to keep an eye on other projects. The only potential issue I see is that some projects are going to be larger than others and will have more mentors. A potential solution could be splitting up some of the students on larger projects into the later small group, but that might not work out very well unless there are regular project meetings.

**Comment by aeksco** [comment]
Yeah it's tough - I think maintaining good communication with everyone in small group is essential, and it's undeniable that large single-project teams disrupt that balance.

I think we should set a threshold that teams of 8 students or more must be split between two different small groups. A team of 7 or fewer could be in a single small group - but any team larger than 8 people must be split up.

Ideally I think we should try to maintain a maximum student-mentor ratio of 4-to-1. This aligns with the threshold above. I.e. a team of eight would require two mentors, so the team would be split into two small groups with one mentor for each. I think this should be a requirement with exceptions granted on a per-case basis.

@kburk1997 @Bad-Science I think this is a good solution. I'm happy to document this and put a pin in it for now. Thoughts?

**Comment by kburk1997** [comment]
Looks like I'm the only returning YACS mentor this semester unless Ada can mentor externally, so if we split up YACS I'll have to do a lot more work outside of small group. It ended up working out somewhat in Fall 2017 when I had to take ECSE seminar, so I'm willing to give it a shot

**Comment by aeksco** [comment]
@kburk1997 Is there any way to trim the YACS team size to be more manageable? How many students look like they're going to be working on it?

**Comment by kburk1997** [comment]
What's worked for us is splitting us up into subteams (backend, frontend, admin). However, in order for this to work the team needs to meet separately outside of RCOS regularly.

**Comment by wdturner** [comment]
Let's put a pin in this until week 2. I believe that Ada intends to mentor,
which would help greatly and we may find that the sub-teams are smaller
this year then last, or maybe we can coerce/develop some more mentors.

**Comment by aeksco** [comment]
Varun, @kburk1997 and I have discussed - if a team is split, it is highly recommended the team as a whole finds opportunities to meet outside of large/small group during the semester to maintain a good collaboration. For large projects the mentors are responsible for coordinating these outside meetings.

**Comment by aeksco** [comment]
Added `backlog` label as per @wdturner's recommendation to put a pin in this until week 2

**Comment by Bad-Science** [comment]
Following up on the mentions here:

Yes, I will be externally mentoring, and after reflecting on things from last year, I think a mandatory outside of rcos meeting is needed for yacs. I'd like to be upfront about this with any potential for credit yacs members, that as part of their minimum commitment, they are required to attend an extra meeting, in person. If they are unable to commit to that then I don't think they should do yacs for credit. I think this will lighten the load on the mentors during small group, and make sure the students we have on yacs are engaged and committed. If there are any other projects that are big enough to split I'd say it's a good idea to have the same requirement.

**Comment by wdturner** [comment]
This may not be a bad thing to recommend for everyone. Your only
communication during the week shouldn't be the small group meetings ...

**Comment by Apexal** [comment]
We might want to revisit this question for Spring 2021 onwards.

**Comment by Apexal** [comment]
Where has the time gone... This should be discussed for Spring **2022** since this semester (Fall 2021) there are many Small Groups with a handful of 1-2 person teams and then huge projects like OpenCircuits, ShuttleTracker, etc. and Mentors are few and far between.

**Comment by Apexal** [comment]
We started to discuss this in our Coordinator meeting and it can be postponed till Project Pairing time.
